### ISSUE TITLE
About gradient penalty

### ISSUE THREAD

**Issue by Aixile** [kodalinaveen3/DRAGAN#2]
Hi, I see that in your paper, the discriminator is updated with

![](http://i.imgur.com/bfJ1dxv.png)

I think the D appeared here means a "discriminator with sigmoid", 
But in the implementation, we always use "tf.nn.sigmoid_cross_entropy_with_logits" instead of explicit calculation of the sigmoid function.

In your paper, I think the gradient penalty is based on "discriminator with sigmoid"
But in your codes, 
`gradients = tf.gradients(discriminator(interpolates), [interpolates])[0]` 
this is the gradient term w.r.t "discriminator without sigmoid"

**Comment by Aixile** [comment]
Constrain the gradient term w.r.t "discriminator with sigmoid" would cause numerical problem.

**Comment by kodalinaveen3** [comment]
We can just use D(x) since sigmoid is monotonic
